src/constraint.rs

```diff
@@ -36,7 +36,7 @@ impl AllOfConstraint {
     pub fn resolve_from_isl_constraint(
         type_references: &[IslTypeRef],
         type_store: &SharedTypeStore,
-        context: &SharedContext,
+        context: &SharedPendingTypes,
```

Comment:
Per our conversation you're planning to simplify this to `&mut PendingTypes` in a future PR. If you don't already have one, turn this comment into an issue to track it.

Reply:
Reference issue: #28